locator/topology.cc

```diff
@@ -438,6 +438,29 @@ const node* topology::find_node(node::idx_type idx) const noexcept {
     return _nodes.at(idx).get();
 }
 
+const node* topology::add_or_update_endpoint(host_id id, std::optional<inet_address> opt_ep, std::optional<endpoint_dc_rack> opt_dr, std::optional<node::state> opt_st, std::optional<shard_id> shard_count)
+{
+    if (tlogger.is_enabled(log_level::trace)) {
+        tlogger.trace("topology[{}]: add_or_update_endpoint: host_id={} ep={} dc={} rack={} state={}, shards={}, at {}", fmt::ptr(this),
+            id, opt_ep.value_or(inet_address{}), opt_dr.value_or(endpoint_dc_rack{}).dc, opt_dr.value_or(endpoint_dc_rack{}).rack, opt_st.value_or(node::state::none), shard_count,
+            current_backtrace());
+    }
+    auto n = find_node(id);
+    if (n) {
+        return update_node(make_mutable(n), std::nullopt, std::move(opt_ep), std::move(opt_dr), std::move(opt_st), std::move(shard_count));
+    } else if (opt_ep && (n = find_node(*opt_ep))) {
+        return update_node(make_mutable(n), std::move(id), std::nullopt, std::move(opt_dr), std::move(opt_st), std::move(shard_count));
+    } else {
+        if (!opt_ep) {
+            on_internal_error(tlogger, format("add_or_update_endpoint: host_id={}: opt_ep must be engaged", id));
+        }
```

Comment:
There is a bug in raft topology, where after decommissioning a node, you're not able to restart nodes because they fail to resolve IPs of left nodes.

Reply:
We need to identify purely by UUID to fix this.